File: implementation/scipamato/scipamato-public-web/src/test/java/ch/difty/scipamato/publ/web/autoconfiguration/BootstrapPropertiesTest.java
package ch.difty.scipamato.publ.web.autoconfiguration;

import static org.assertj.core.api.Assertions.assertThat;

import de.agilecoders.wicket.themes.markup.html.bootswatch.BootswatchTheme;
import org.junit.Test;

public class BootstrapPropertiesTest {

    private final BootstrapProperties bp = new BootstrapProperties();

    @Test
    public void standardTheme_isSandstone() {
        assertThat(bp.getTheme()).isEqualTo(BootswatchTheme.Sandstone);
    }

    @Test
    public void canModifyTheme() {
        bp.setTheme(BootswatchTheme.Cerulean);
        assertThat(bp.getTheme()).isEqualTo(BootswatchTheme.Cerulean);
    }

    @Test
    public void isEnabledByDefault() {
        assertThat(bp.isEnabled()).isTrue();
    }

    @Test
    public void canDisable() {
        bp.setEnabled(false);
        assertThat(bp.isEnabled()).isFalse();
    }

}

File: public/public-web/src/test/kotlin/ch/difty/scipamato/publ/web/autoconfiguration/BootstrapPropertiesTest.kt
package ch.difty.scipamato.publ.web.autoconfiguration

import de.agilecoders.wicket.themes.markup.html.bootswatch.BootswatchTheme
import org.amshove.kluent.shouldBeEqualTo
import org.amshove.kluent.shouldBeFalse
import org.amshove.kluent.shouldBeTrue
import org.junit.jupiter.api.Test

class BootstrapPropertiesTest {

    private val bp = BootstrapProperties()

    @Test
    fun standardTheme_isSandstone() {
        bp.theme shouldBeEqualTo BootswatchTheme.Sandstone
    }

    @Test
    fun canModifyTheme() {
        bp.copy(theme = BootswatchTheme.Cerulean).run {
            theme shouldBeEqualTo BootswatchTheme.Cerulean
        }
    }

    @Test
    fun isEnabledByDefault() {
        bp.isEnabled.shouldBeTrue()
    }

    @Test
    fun canDisable() {
        bp.copy(isEnabled = false).run {
            isEnabled.shouldBeFalse()
        }
    }
}
